src/lambda_function/s3_event_config/app.py

```diff
@@ -40,14 +48,111 @@ def lambda_handler(event, context):
         raise KeyError(err_msg)
 
 
+def get_existing_bucket_notification_configuration_and_type(
+    s3_client: boto3.client, bucket: str, destination_type: str
+) -> typing.Tuple[dict, list]:
+    """
+    Gets the existing bucket notification configuration and the existing notification
+    configurations for a specific destination type.
+
+    Arguments:
+        s3_client (boto3.client) : s3 client to use for s3 event config
+        bucket (str): bucket name of the s3 bucket to delete the config in
+        destination_type (str): String name of the destination type for the configuration
+
+    Returns:
+        Tuple: A bucket notifiction configuration,
+            and the notification configurations for a specific destination type
+    """
+    existing_bucket_notification_configuration = (
+        s3_client.get_bucket_notification_configuration(Bucket=bucket)
+    )
+
+    # Remove ResponseMetadata because we don't want to persist it
+    existing_bucket_notification_configuration.pop("ResponseMetadata", None)
+
+    existing_notification_configurations_for_type = (
+        existing_bucket_notification_configuration.get(
+            f"{destination_type}Configurations", []
+        )
+    )
+
+    # Initialize this with an empty list to have consistent handling if it's present
+    # or missing.
+    if not existing_notification_configurations_for_type:
+        existing_bucket_notification_configuration[
+            f"{destination_type}Configurations"
+        ] = []
+
+    return (
+        existing_bucket_notification_configuration,
+        existing_notification_configurations_for_type,
+    )
+
+
+def get_matching_notification_configuration(
+    destination_type_arn: str,
+    existing_notification_configurations_for_type: list,
+    destination_arn: str,
+) -> typing.Union[tuple[int, dict], tuple[None, None]]:
```

Comment:
Likewise, this would be more readable as a named tuple or dict.

Reply:
I added this in https://github.com/Sage-Bionetworks/recover/pull/87/commits/f7c2e3a5613aae43c90a9cd0300e1995aa9b96e6